eval/src/vespa/eval/instruction/dense_single_reduce_function.cpp

```diff
@@ -220,14 +221,16 @@ DenseSingleReduceFunction::DenseSingleReduceFunction(const DenseSingleReduceSpec
       _inner_size(spec.inner_size),
       _aggr(spec.aggr)
 {
+    assert(result_type().cell_meta().is_scalar == false);
 }
 
 DenseSingleReduceFunction::~DenseSingleReduceFunction() = default;
 
 InterpretedFunction::Instruction
 DenseSingleReduceFunction::compile_self(const ValueBuilderFactory &, Stash &stash) const
 {
-    auto op = typify_invoke<4,MyTypify,MyGetFun>(result_type().cell_type(), _aggr,
+    auto op = typify_invoke<4,MyTypify,MyGetFun>(result_type().cell_meta().limit().not_scalar(),
```

Comment:
use input type with not_scalar and infer the output cell type in GetMyFun to support unstable cell types
(short term work-around could be to require that input and result cell types are equal)